### PR TITLE
Add Conformance tests

### PR DESCRIPTION
**Pull request by cpanato**
**What this PR does / why we need it**:
Add conformance tests to CAPDO

Before merging this PR we need to merge: https://github.com/kubernetes/test-infra/pull/19774

- Also update CAPI to 0.3.10

/hold

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes-sigs/cluster-api-provider-digitalocean/issues/195

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
Add k8s conformance tests to CAPDO
```

### CODE REVIEW

**Comment by cpanato** [comment]
override the e2e to run conformance to validate: here is the job: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-digitalocean/201/pull-cluster-api-provider-digitalocean-e2e/1322097670017257472
🎉

File: scripts/ci-e2e.sh [hunk 1]
```diff
@@ -56,4 +56,5 @@ trap 'remove_ssh_key ${SSH_KEY_FINGERPRINT}' EXIT
 export DO_SSH_KEY_FINGERPRINT=${SSH_KEY_FINGERPRINT}
 
 make test-e2e
+
```
Comment (MorrisLaw) [inline scripts/ci-e2e.sh:59]: Extra whitespace?

Comment (cpanato) [reply scripts/ci-e2e.sh:59]: oops :)

Comment (cpanato) [reply scripts/ci-e2e.sh:59]: I will wait for the conformance test to finish to fix this, ok?

**Comment by cpanato** [comment]
conformance rans fine: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-digitalocean/201/pull-cluster-api-provider-digitalocean-conformance-v1alpha3/1322168531755732992

**Comment by cpanato** [comment]
will not retrigger the conformance and e2e again, the update was just removing blank lines.

ready for review @MorrisLaw @prksu

**Pull request merged**